### ISSUE TITLE
Move call type implementations to new package calltype

### ISSUE THREAD

**Issue by meling** [relab/gorums#109]
Moving the call types to a new package `calltype` can help with code maintainability. @Raytar Thoughts?

**Comment by Raytar** [comment]
Yes, that sounds like a good idea. It would also help with making the API clearer since the call types are meant to be used by generated code.
